lib/portlayer/attach/communication/interactor.go

```diff
@@ -232,9 +235,13 @@ func (t *interaction) Unblock() error {
 		ok, err := t.channel.SendRequest(msgs.UnblockReq, true, []byte(msgs.UnblockMsg))
 		if !ok || err != nil {
 			log.Errorf("failed to unblock the other side: %s", err)
-			t.unblocked.Reset()
+			// #5038: resync package is not reentrant so we need to call Reset after this
+			atomic.StoreUint32(&reset, 1)
 		}
 	})
 
+	if atomic.LoadUint32(&reset) == 1 {
```

Comment:
yeah now that you pointed out this could be a non-atmomic var., refreshed the pr